### DISCUSSION TITLE
Would it be possible to add a translation and dictionary system?

### DISCUSSION THREAD

**Discussion by Pretinaverse** [edrlab/thorium-reader#2615] [Ideas]
I was wondering if it would be possible to integrate Google Translate or Deepl into Thorium.

Most of the books I read are in English, but my first language is Spanish. So from time to time I come across words I don't know.
**It would be great if you could select a word, right click on it and get an option to translate it or look it up in the dictionary, and when you click on it, a pop-up window would appear with the translation or the meaning, depending on which option you chose.**

This would not only help those who read books in a language other than their own, but also those who read in their native language but still come across an unfamiliar word from time to time.

**Comment by danielweck** [comment]
The (small) development team is currently focussing on the annotations feature, which already exists but we are incrementally implementing extensions such as import/export based on an interchange format that will work not only between Thorium users but also other reading system apps / solutions.

**Reply by danielweck** [reply]
... beyond the current development priorities, it absolutely makes sense to aim to offer dictionary lookup and translation features. Maybe even LLM / AI / ChatGPT -style interactive prompting ;)

**Comment by Velo-C** [comment]
If you are also reading on a computer, I currently use the built-in functionality of the desktop version of DeepL directly. It allows you to use shortcuts to bring up a small window that translates the currently selected passage. It's kind of an emergency solution. Of course I'd love to see more integration, I really like this reader!

**Reply by danielweck** [reply]
Hello, are you saying the Deepl is already capable of reading text inside Thorium, or that Deepl offers its own reading view?

**Reply by Velo-C** [reply]
Oh, hi! I thought I'd use a screenshot to make it clearer. DeepL The desktop version of the program has an option that, when opened, can be triggered by a shortcut key. For example, what I did was to select the text, then press the default Ctrl+C+C, and this little hovering window will pop up!
![1225tran](https://github.com/user-attachments/assets/a630d95c-628c-4eb4-a6d2-997f77d27602)

**Reply by yumanuralfath** [reply]
thanks for info, this what i need
